### DISCUSSION TITLE
How to define our own routes with advanced configuration?

### DISCUSSION THREAD

**Discussion by vince83110** [nuxt/nuxt#17870] [Questions]
In Nuxt2, using `nuxt/i18n`, I was able to change the name of paths to match any of my file in the `pages` folder.

It was not perfect, but it allowed me to handle my language (_french_) correctly, I don't want to have a folder _produits_ (products in english) in my code, just to have the path with `produits`, like : 

`www.mysite.fr/produits/my-slug
`

For now, on Nuxt 3 I use this way, I change folder's names but there are so many limits.

How to handle correctly routing in Nuxt 3? I looked at `vue-router` documentation, but I can't see how to create routes from a config file.

I want to define my routes not according to my folder names, but according to more advanced configuration.

Is it possible?

**Comment by vincenthalman** [comment]
I don't know if I fully understand you. In my project I also wanted to change some routing configuration and add some routes dynamic. In your case, what you could do is to add the routes dynamically so you are free to adjust the route configs.

use the addRoute() for example:

/layouts/default.vue

`<script setup>`
`const router = useRouter();`
`router.addRoute({path: '/example', name: 'examplename', component: () => import ("~/yourdirectory/example.vue") })` 
`</script>`

**Reply by vince83110** [reply]
It seems to be what I want to achieve, I just want to use configuration to set my paths, I don't want to rely on folder names.

I will try to put it in kind of configuration file as JSON and have something clean, I need to find a way to disable this routing by folder too. I let open, will try today and give feedback.

Thanks for answer, nice name btw !

**Comment by lacorde** [comment]
@vincenthalman Thank you for your answer !

Maybe you or anyone can help me further with this.

Here is my **pages** folder structure :

```
pages/
 category/
  [slug.vue]
 product/
  [slug.vue]
```

I would like to be able to access a page - for example [website.com/category/category1](url) - directly at [website.com/category1](url).
In short, **being able to access a product or a category directly using the slug**.

Is there any way to achieve this behavior using Nuxt 3 ?

**Reply by lacorde** [reply]
I tried adding route aliases in script setup as suggested, but this only work client-side (not loading  [website.com/category1](url) directly).

I tried adding routes server side without success (maybe my method isn't quite right) :
```
export default defineNuxtPlugin(({ ssrContext }) => {
  const router = useRouter();
  if (process.server && ssrContext) {
    router.removeRoute("categorie-slug"); // remove default route
    router.addRoute({ // add custom route with alias
      name: "categorie-slug",
      path: "/categorie/:slug",
      component: () => import("/_nuxt/pages/categorie/[slug].vue"),
      alias: "/:slug",
    });
  }
```

**Reply by qnp** [reply]
@lacorde I would suggest you to use `extendPages` in a nuxt module, as such in [this example](https://github.com/nuxt/framework/blob/main/examples/module-extend-pages/modules/pages/index.ts)
